sumologic/resource_sumologic_generic_polling_source.go

```diff
@@ -25,18 +25,19 @@ func resourceSumologicGenericPollingSource() *schema.Resource {
 		Required: true,
 		ForceNew: true,
 		ValidateFunc: validation.StringInSlice([]string{"AwsS3Bucket", "AwsElbBucket", "AwsCloudFrontBucket",
-			"AwsCloudTrailBucket", "AwsS3AuditBucket", "AwsCloudWatch", "AwsInventory", "AwsXRay", "GcpMetrics", "AwsS3ArchiveBucket"}, false),
+			"AwsCloudTrailBucket", "AwsS3AuditBucket", "AwsCloudWatch", "AwsInventory", "AwsXRay", "GcpMetrics", "AwsS3ArchiveBucket", "AzureEventHubLog"}, false),
 	}
 	pollingSource.Schema["scan_interval"] = &schema.Schema{
 		Type:     schema.TypeInt,
-		Required: true,
+		Optional: true,
```

Comment:
Need to provide a default value.

Reply:
I thought by keeping the hardcoded setting for event hub source we wouldn't need to change the scan interval from required to optional? https://github.com/SumoLogic/terraform-provider-sumologic/pull/626/files#diff-3cd6cf5332a73badbeed5ade27b5168f076cd22222dd45f273384d746dfb1c21R349

Reply:
Yeah even we keep the hardcoded setting, we have to make it optional otherwise customers are required to set scan_interval for azure event hub log sources. 
<img width="1419" alt="Screenshot 2024-05-09 at 8 39 13 PM" src="https://github.com/SumoLogic/terraform-provider-sumologic/assets/124813412/6afaeded-22c2-4172-b69b-03f044eb31a5">

Reply:
Any need to modify the polling source's terraform documentation while we change `scan_interval` to optional?